cms/static/js/spec/views/modals/move_xblock_modal_spec.js

```diff
@@ -44,9 +43,209 @@ define(['jquery', 'underscore', 'edx-ui-toolkit/js/utils/spec-helpers/ajax-helpe
+                    getConfirmationFeedbackTitleLink(targetParentLocator)
+                );
+                expect(modal.movedAlertView.options.messageLink).toEqual(
+                    getConfirmationFeedbackMessageLink(
+                        sourceDisplayName,
+                        sourceLocator,
+                        sourceParentLocator,
+                        sourceIndex
+                    )
+                );
+            };
+
+            it('moves an xblock when move button is clicked', function() {
+                var requests = AjaxHelpers.requests(this);
+                moveXBlockWithSuccess(requests);
+            });
+
+            it('undo move an xblock when undo move button is clicked', function() {
+                var sourceIndex = 0,
+                    requests = AjaxHelpers.requests(this);
+                moveXBlockWithSuccess(requests);
+                modal.movedAlertView.undoMoveXBlock({
+                    target: $(modal.movedAlertView.options.messageLink.text)
+                });
+                AjaxHelpers.respondWithJson(requests, {
+                    move_source_locator: sourceLocator,
+                    parent_locator: sourceParentLocator,
+                    target_index: sourceIndex
+                });
+                expect(modal.movedAlertView.movedAlertView.options.title).toEqual(
+                    getUndoConfirmationFeedbackTitle(sourceDisplayName)
+                );
+            });
+
+            it('does not move an xblock when cancel button is clicked', function() {
+                var sourceIndex = 0;
+                // select a target parent and click cancel button
+                selectTargetParent(targetParentLocator);
+                modal.$el.find('.modal-actions .action-cancel')[sourceIndex].click();
+                expect(modal.movedAlertView).toBeNull();
+            });
+
+            it('shows a notification when moving', function() {
+                var requests = AjaxHelpers.requests(this),
+                    notificationSpy = ViewHelpers.createNotificationSpy();
+                // select a target item and click on move
+                selectTargetParent(targetParentLocator);
+                modal.$el.find('.modal-actions .action-move').click();
+                verifyNotificationStatus(requests, notificationSpy, 'Moving');
+            });
+
+            it('shows a notification when undo moving', function() {
+                var notificationSpy,
+                    requests = AjaxHelpers.requests(this);
+                moveXBlockWithSuccess(requests);
+                notificationSpy = ViewHelpers.createNotificationSpy();
+                modal.movedAlertView.undoMoveXBlock({
+                    target: $(modal.movedAlertView.options.messageLink.text)
+                });
+                verifyNotificationStatus(requests, notificationSpy, 'Undo moving');
+            });
```

Comment:
@mushtaqak can we confirm the developer error messages via test cases e.g invalid move. I know they're not user facing but i think it's worth to write 1 or 2 test(s) for that.